### PR TITLE
Remove default null-count parameter from cudf::make_strings_column factory

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Removes the `UNKNOWN_NULL_COUNT` default parameter from `cudf::make_strings_column` factory functions and updates the callers to always pass the null-count.

Labeling this with breaking since it is a public function.

Contributes to: https://github.com/rapidsai/cudf/issues/11968

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/benchmarks/string/factory.cu [hunk 1]
```diff
@@ -63,7 +63,7 @@ static void BM_factory(benchmark::State& state)
 
   for (auto _ : state) {
     cuda_event_timer raii(state, true, cudf::get_default_stream());
-    cudf::make_strings_column(pairs);
+    cudf::make_strings_column(pairs, cudf::get_default_stream());
```
Comment (karthikeyann) [inline cpp/benchmarks/string/factory.cu:66]: will previous thrust call on `thrust::device` use the same stream as `cudf::get_default_stream()` ?

Comment (karthikeyann) [reply cpp/benchmarks/string/factory.cu:66]: The default value on the `stream` argument does not seem to be removed in this PR. should it be included too?

Comment (davidwendt) [reply cpp/benchmarks/string/factory.cu:66]: No, removing default values only applies to detail functions. We'll be addressing the stream parameter in public APIs like this one later.

**Review by harrism — APPROVED** [review]

**Review by karthikeyann — APPROVED** [review]

**Pull request merged**